packages/edit-post/src/components/header/fullscreen-mode-close/index.js

```diff
@@ -7,12 +7,23 @@ import { get } from 'lodash';
  * WordPress dependencies
  */
 import { useSelect } from '@wordpress/data';
-import { Button } from '@wordpress/components';
+import { useEntityProp } from '@wordpress/core-data';
+import { Button, Icon } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 import { addQueryArgs } from '@wordpress/url';
 import { wordpress } from '@wordpress/icons';
 
 function FullscreenModeClose() {
+	const [ siteIconURL ] = useEntityProp( 'root', 'site', 'site_icon_url' );
```

Comment:
I think useEntityProp is meant to be used in conjunction with a context provider around the block editor (so more for blocks) or is this a correct usage @epiqueras ?

Why not just `wp.data.select('core').getSite()` in the useSelect

Reply:
also, should we add this API request to the preloaded paths?

Reply:
Yeah, we should use `getEntityRecords( 'root', 'site' ).site_icon_url` to make it simpler and decoupled from any providers.

Reply:
interesting :) Is this documented behavior? It's a confusing difference IMO

Reply:
> should we add this API request to the preloaded paths?

@youknowriad Would you be able to elaborate more on what it means to add this to preloaded paths?

Reply:
yes, basically if you add a path to this array, the server will inject  the response on the initial page load avoiding to trigger a request that we  know will always be triggered on that page.

A similar array exists for edit-post too.

Reply:
[As suggested by Timothy](https://github.com/WordPress/gutenberg/pull/22952#discussion_r438124552), we now use the WordPress REST index to register site icon url. It looks like the index is already being preloaded [here](https://github.com/WordPress/gutenberg/blob/master/lib/edit-site-page.php#L190), but let me know if there's more to do.